src/test/java/de/tum/cit/aet/artemis/iris/PyrisConcistencyCheckIntegrationTest.java

```diff
@@ -0,0 +1,134 @@
+import de.tum.cit.aet.artemis.iris.service.IrisConsistencyCheckService;
+import de.tum.cit.aet.artemis.iris.service.pyris.PyrisDTOService;
+import de.tum.cit.aet.artemis.iris.service.pyris.dto.consistencyCheck.PyrisConsistencyCheckStatusUpdateDTO;
+import de.tum.cit.aet.artemis.iris.service.pyris.dto.data.PyrisProgrammingExerciseDTO;
+import de.tum.cit.aet.artemis.iris.service.pyris.dto.status.PyrisStageDTO;
+import de.tum.cit.aet.artemis.iris.service.pyris.dto.status.PyrisStageState;
+import de.tum.cit.aet.artemis.iris.service.pyris.job.ConsistencyCheckJob;
+import de.tum.cit.aet.artemis.programming.domain.ProgrammingExercise;
+import de.tum.cit.aet.artemis.programming.domain.ProgrammingLanguage;
+
+@Profile(PROFILE_IRIS)
+class PyrisConsistencyCheckIntegrationTest extends AbstractIrisIntegrationTest {
+
+    private static final String TEST_PREFIX = "pyrisconsistencytest";
+
+    @Autowired
+    private UserUtilService userUtilService;
+
+    @Mock
+    private PyrisDTOService pyrisDTOService;
+
+    @Autowired
+    private IrisConsistencyCheckService irisConsistencyCheckService;
+
+    private Course course;
+
+    private ProgrammingExercise programmingExercise;
+
+    @BeforeEach
+    void initTestCase() {
+        userUtilService.addUsers(TEST_PREFIX, 2, 1, 0, 2);
+        course = programmingExerciseUtilService.addCourseWithOneProgrammingExercise();
+        this.programmingExercise = course.getExercises().stream().filter(ProgrammingExercise.class::isInstance).map(ProgrammingExercise.class::cast).findFirst().orElse(null);
+
+        userUtilService.createAndSaveUser(TEST_PREFIX + "student42");
+        userUtilService.createAndSaveUser(TEST_PREFIX + "tutor42");
+        userUtilService.createAndSaveUser(TEST_PREFIX + "instructor42");
+        activateIrisFor(course);
+    }
+
+    @Test
+    @WithMockUser(username = TEST_PREFIX + "instructor1", roles = "INSTRUCTOR")
+    void callConsistencyCheckAsInstructor_shouldSucceed() throws Exception {
+        PyrisProgrammingExerciseDTO exerciseDto = createPyrisProgrammingExerciseDTO(programmingExercise);
+        irisRequestMockProvider.mockProgrammingConsistencyCheckResponse(dto -> {
+            assertThat(dto.exercise().id()).isEqualTo(exerciseDto.id());
+            assertThat(dto.exercise().name()).isEqualTo("Programming");
+            assertThat(dto.exercise().programmingLanguage()).isEqualTo(ProgrammingLanguage.JAVA);
+        }, exerciseDto.id());
+
+        when(pyrisDTOService.toPyrisProgrammingExerciseDTO(any())).thenReturn(exerciseDto);
+
+        request.postWithoutResponseBody("/api/iris/consistency-check/exercises/" + exerciseDto.id(), exerciseDto, HttpStatus.OK);
+
+        // in the normal system, at some point we receive a websocket message with the result
+
+        List<PyrisStageDTO> stages = List.of(new PyrisStageDTO("Generating Consistency Check", 10, PyrisStageState.DONE, null));
+        String jobId = "testJobId";
```

Comment:
This belongs to the block below, please group it accordingly
```suggestion
        // in the normal system, at some point we receive a websocket message with the result
        List<PyrisStageDTO> stages = List.of(new PyrisStageDTO("Generating Consistency Check", 10, PyrisStageState.DONE, null));
        String jobId = "testJobId";
```